R/data_modify.R

```diff
@@ -352,8 +356,12 @@ data_modify.grouped_df <- function(data, ..., .if = NULL, .at = NULL, .modify =
   # finally, we can evaluate expression and get values for new variables
   symbol_string <- insight::safe_deparse(symbol)
   if (!is.null(symbol_string) && all(symbol_string == "n()")) {
-    # "special" functions
+    # "special" functions - using "n()" just returns number of rows
     new_variable <- nrow(data)
+  } else if (!is.null(symbol_string) && length(symbol_string) == 1 && grepl("n()", symbol_string, fixed = TRUE)) {
```

Comment:
This feels very hacky and it seems it would break if someone passes `1:fun()` since `n()` will be detected. I also think it would open the door to more "special" functions (`first()`, `last()`, `nth()`, etc.), which can be hard to maintain in the long run.

Random thought: would parsing the expression with getParseData() and using its results make this more robust?

Reply:
ok, should no be catching only "n()".

> I also think it would open the door to more "special" functions 

True, but we already have `n()` in `data_summary()`, and I think it's a more common use case than first/last. Thus, this is more a "feature completion" than opening Pandorra's box ;-)

What do you mean by `getParseData()`?